packages/hardhat-chai-matchers/src/bigNumber.ts

```diff
@@ -134,6 +136,33 @@ function overwriteBigNumberFunction(
         expected,
         actual
       );
+    } else if (functionName === "eq" && chaiUtils.flag(this, "deep")) {
+      // this is close enough to what chai itself does, except we compare
+      // numbers after normalizing them
+      const comparator = (a: any, b: any): boolean | null => {
+        try {
+          const normalizedA = normalizeToBigInt(a);
+          const normalizedB = normalizeToBigInt(b);
+          return normalizedA === normalizedB;
+        } catch (e) {
+          // use default comparator
+          return null;
+        }
+      };
+
+      const prevLockSsfi = chaiUtils.flag(this, "lockSsfi");
+      chaiUtils.flag(this, "lockSsfi", true);
```

Comment:
wtf is `lockSsfi`?

Reply:
This is part of the stuff I just copied from chai. From what I understand, they do this to have better stack traces:

> ssfi: start stack function - used to prevent callback stacks from being shown in errors.

I will add a comment clarifying this.

Reply:
Done. Let me know if the comment makes sense.